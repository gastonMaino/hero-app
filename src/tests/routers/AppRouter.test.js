import { mount } from "enzyme"
import { AuthContext } from "../../auth/AuthContext"
import { AppRouter } from "../../routers/AppRouter"

describe('test in <AppRouter />', () => {

    const contexValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }

    test('should show login if it is no authenticated', () => {
        const wrapper = mount(
            <AuthContext.Provider value={contexValue}>
                <AppRouter />
            </AuthContext.Provider>
        )

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('PublicRoutes').exists()).toBe(true);
        expect(wrapper.find('PrivateRoutes').exists()).toBe(false);
    })

    test('should show marvel component if is authenticated', () => {

        const contexValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'juan'
            }
        }


        const wrapper = mount(
            <AuthContext.Provider value={contexValue}>
                <AppRouter />
            </AuthContext.Provider>
        )

        expect(wrapper.find('.main-menu').exists()).toBe(true);
    })
    
})
