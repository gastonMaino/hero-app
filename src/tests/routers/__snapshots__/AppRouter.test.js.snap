// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <AppRouter /> should show login if it is no authenticated 1`] = `
<div>
  <main
    className="main-login"
  >
    <article
      className="container"
    >
      <div
        className="container-title-login"
      >
        <h1
          className="title-login"
        >
          Login Screen
        </h1>
      </div>
      <button
        className="button-primary-login"
        onClick={[Function]}
      >
        Login
      </button>
    </article>
  </main>
</div>
`;
